packages/supi/src/install/index.ts

```diff
@@ -401,6 +403,41 @@ export async function mutateModules (
       })
     }
 
+    if (maybeOpts.strict) {
+      const project = projectsToInstall[0]
+      let dependenciesList: string[] = []
+      switch (project.mutation) {
+        case 'installSome': {
+          if (project.allowNew === false) {
+            for (const p of projectsToInstall) {
+              dependenciesList = [...p.wantedDependencies.map(wantedDependency => wantedDependency.raw), ...dependenciesList]
+            }
+            for (const input of project.dependencySelectors) {
+              if (!dependenciesList.includes(input)) {
+                throw new PnpmError('NO_PACKAGE_IN_DEPENDENCY', `No ${input} package found in dependencies of the project`)
+              }
+            }
```

Comment:
I think we should make it less strict. If at least one of the selectors is found, then just print an info message about the not found selectors. If none of the selectors is found then throw an error.